src/libs/ReportUtils.ts

```diff
@@ -5144,10 +5144,11 @@ function hasUpdatedTotal(report: OnyxEntry<Report>): boolean {
  * - The action is listed in the thread-disabled list
  * - The action is a split bill action
  * - The action is deleted and is not threaded
+ * - The report is archived and the action is not threaded
  * - The action is a whisper action and it's neither a report preview nor IOU action
  * - The action is the thread's first chat
  */
-function shouldDisableThread(reportAction: OnyxEntry<ReportAction>, reportID: string): boolean {
+function shouldDisableThread(reportAction: OnyxEntry<ReportAction>, reportID: string, isArchivedReport = false): boolean {
```

Comment:
You don't need to add a new method parameter for `isArchivedReport`. You can check if the report is archived in the `shouldDisableThread` method using the following code:

```diff
+const report = getReport(reportID);
+const isArchivedReport = isArchivedRoom(report);
```

Reply:
Since it's a utility method, let's keep it self-contained. You will also notice that the `getReport` method is used by other methods in the file.